qadi_simple_multi.py

```
@@ -188,66 +197,119 @@
+        # Show all non-empty lines if no I: prefix found
+        for line in patterns.split('\n'):
+            if line.strip():
+                print(f"  • {line.strip()}")
     
     # Final synthesis
-    print("\n✨ SYNTHESIS:")
-    print("-" * 70)
+    render_section_header("SYNTHESIS", "✨")
+    
+    # Smart summarization to preserve key insights while fitting context window
+    def extract_key_insights(text, max_chars=600):
```

Ruff (0.12.2)

Missing return type annotation for private function `extract_key_insights`

(ANN202)